rendermodules/montage/run_montage_job_for_section.py

```diff
@@ -167,8 +102,22 @@ def __init__(self, schema_type=None, *args, **kwargs):
         # Assigning the solver_executable to a different variable and removing it from args
         self.solver_executable = self.args['solver_executable']
         self.args.pop('solver_executable', None)
+        self.clone_section_stack = self.args['clone_section_stack']
+        self.args.pop('clone_section_stack')
+        if (self.args['first_section']!=self.args['last_section']):
```

Comment:
this condition should be applied only when the clone_section_stack is set to true...

Reply:
i don't think that's right... this part needs to pop the parameter from the arguments, it has a value no matter what even if it's not set because i gave it a default in the schema.

Reply:
Yes, I agree with what you said... But I meant the "if" condition that checks for the self.args['first_section']!= self.args['last_section']. 
This condition is checked regardless of the value of "clone_section_stack". However, it should not be unless you are running this code for only one section. This might not be the case if you want to montage a range of sections  (for the subsequent time with different parameters using the source stack with "COMPLETE" state).

encapsulating this "if" condition (line 107) with another one that checks if "clone_section_stack" is set to true would be appropriate.

Reply:
ah! yes i see what you are saying now. You are right! i'll change this.